lib/services/azuresqldb/cmd-provision.js

```diff
@@ -230,10 +234,14 @@ var sqldbProvision = function (log, params) {
                                 callback(null);
                             }
                         });
-                    } else {  // sql server exists
+                    } else if (result.statusCode === HttpStatus.OK) {  // sql server exists
```

Comment:
Wouldn't the logic be cleaner (and saving a few lines of code) if we do getServer and check for OK status first?  That's always required anyway.  Only when getServer fails would we need to worry about if we have privilege to create server.